docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,172 @@
+```shell
+clusterctl alpha topology plan -f new-cluster-and-class.yaml -o output/
+```
+Produces an output similar to this:
+```shell
+Detected a cluster with Cluster API installed. Will use it to fetch missing objects.
+The following ClusterClasses will be affected by the changes:
+ ＊ default/new-cluster-class
+
+The following Clusters will be affected by the changes:
+ ＊ default/new-cluster
+
+Changes for Cluster "default/new-cluster": 
+
+  NAMESPACE  KIND                   NAME                              ACTION    
+  default    DockerCluster          new-cluster-jfjcs                 created   
+  default    DockerMachineTemplate  new-cluster-control-plane-6rkg8   created   
+  default    DockerMachineTemplate  new-cluster-md-0-infra-4nv62      created   
+  default    DockerMachineTemplate  new-cluster-md-1-infra-dcfnj      created   
+  default    KubeadmConfigTemplate  new-cluster-md-0-bootstrap-kk24t  created   
+  default    KubeadmConfigTemplate  new-cluster-md-1-bootstrap-k6xw4  created   
+  default    KubeadmControlPlane    new-cluster-dtdvx                 created   
+  default    MachineDeployment      new-cluster-md-0-z9v9f            created   
+  default    MachineDeployment      new-cluster-md-1-xjtg2            created   
+  default    Secret                 new-cluster-shim                  created   
+  default    Cluster                new-cluster                       modified  
+
+Created objects are written to directory "output/created"
+Modified objects are written to directory "output/modified"
+
+```
+
+The final state of objects that are created and updated are written to disk under the `output` folder. For objects that are modified the following
+files are written to disk:
+* Original object
+* Final object
+* JSON patch between the original and the final object
+* Diff of the original and final objects
+
+The contents fn the output directory is similar to this:
+```
+output
+├── created
+│   ├── DockerCluster_default_new-cluster-jfjcs.yaml
+│   ├── DockerMachineTemplate_default_new-cluster-control-plane-6rkg8.yaml
+│   ├── DockerMachineTemplate_default_new-cluster-md-0-infra-4nv62.yaml
+│   ├── DockerMachineTemplate_default_new-cluster-md-1-infra-dcfnj.yaml
+│   ├── KubeadmConfigTemplate_default_new-cluster-md-0-bootstrap-kk24t.yaml
+│   ├── KubeadmConfigTemplate_default_new-cluster-md-1-bootstrap-k6xw4.yaml
+│   ├── KubeadmControlPlane_default_new-cluster-dtdvx.yaml
+│   ├── MachineDeployment_default_new-cluster-md-0-z9v9f.yaml
+│   ├── MachineDeployment_default_new-cluster-md-1-xjtg2.yaml
+│   └── Secret_default_new-cluster-shim.yaml
+└── modified
+    ├── Cluster_default_new-cluster.diff
+    ├── Cluster_default_new-cluster.jsonpatch
+    ├── Cluster_default_new-cluster.modified.yaml
+    └── Cluster_default_new-cluster.original.yaml
+```
+
+### Modifying a ClusterClass that affects multiple Clusters
+
+```shell
+clusterctl alpha topology plan -f modified-first-cluster-class.yaml -o output/
```

Comment:
```suggestion
clusterctl alpha topology plan -f modified-cluster-class.yaml -o output/
```
nit, I think for the purpose of this doc this is slightly better (I would also drop the first in the clusterclass name in the output)

Reply:
Also in this case, let's link to the example, and document what is expected to be in the file